Entities/Vehicles/Ballista/Ballista.as

```diff
@@ -314,27 +310,32 @@ bool Vehicle_canFire(CBlob@ this, VehicleInfo@ v, bool isActionPressed, bool was
 
 void Vehicle_onFire(CBlob@ this, VehicleInfo@ v, CBlob@ bullet, const u8 _charge)
 {
+	u16 delay = cooldown_time;
+
 	if (bullet !is null)
 	{
 		u8 charge_prop = _charge;
 
-		f32 charge = 5.0f + 15.0f * (float(charge_prop) / float(v.max_charge_time));
+		f32 charge = 5.0f + 15.0f * (float(charge_prop) / float(v.getCurrentAmmo().max_charge_time));
 
 		f32 angle = getAngle(this, _charge, v) + this.getAngleDegrees();
 		Vec2f vel = Vec2f(0.0f, -charge).RotateBy(angle);
 		bullet.setVelocity(vel);
 		bullet.setPosition(bullet.getPosition() + vel);
 
-		if (this.get_bool("bomb ammo"))
+		bool bomb_bolts_selected = v.getCurrentAmmo().ammo_name == "mat_bomb_bolts";
```

Comment:
It would be better if the ammo had a property for the cooldown time instead of hardcoding it all.

Reply:
It does have a property - that's AmmoInfo.fire_delay (unless you mean something else). 

From what I understand, the original VehicleCommon.as had v.fire_delay and v.cooldown_time.  Fire_delay is set when we make a vehicle (and we do that now too for a.fire_delay). In the case of ballistas, fire_delay for each ammo does not change. Whenever a shot is made, fire_delay is used to set v.fire_time to the gametime when we'll be able to start firing a shot again.

```
void SetFireDelay(CBlob@ this, int shot_delay, VehicleInfo@ v)
{
	v.firing = false;
	v.fire_time = (getGameTime() + shot_delay);
}
```

cooldown_time is the amount of time a vehicle has to wait (updated every tick) before it's able to fire any ammo again. Since it's not an ammo-specific delay and applies to the entire vehicle at that time, I thought it should be left like that. In the original files cooldown_time is also modified like this. cooldown_time is mostly used for displaying the red cooldown bar after you fire a shot.

I don't really know how to modify it more if it's still necessary and maybe I'm missing something obvious, I don't know

Reply:
I'm struggling to understand vehicle code so I may also be missing something, but all I'm suggesting is to use fire_delay instead of the script variables. My current understanding tells me this should work:
```angelscript
void Vehicle_onFire(CBlob@ this, VehicleInfo@ v, CBlob@ bullet, const u8 _charge)
{
	if (bullet !is null)
	{
		u8 charge_prop = _charge;

		f32 charge = 5.0f + 15.0f * (float(charge_prop) / float(v.getCurrentAmmo().max_charge_time));

		f32 angle = getAngle(this, _charge, v) + this.getAngleDegrees();
		Vec2f vel = Vec2f(0.0f, -charge).RotateBy(angle);
		bullet.setVelocity(vel);
		bullet.setPosition(bullet.getPosition() + vel);

		bool bomb_bolts_selected = v.getCurrentAmmo().ammo_name == "mat_bomb_bolts";

		if (bomb_bolts_selected)
		{
			bullet.Tag("bomb ammo");
			bullet.Sync("bomb ammo", true);
		}
	}

	v.last_charge = _charge;
	v.charge = 0;
	v.cooldown_time =  v.getCurrentAmmo().fire_delay; // <------ this is what I am referring to
}
```
I'm also just realising you would still need to hardcode the tagging code so this is pretty much all you can do.

Reply:
Done, I guess I misunderstood it, is there anything else to do or can this be resolved?